### ISSUE TITLE
Using Mbi6033 with WLED (Blinking and Empty Data problem) 

### ISSUE THREAD

**Issue by seyfettinkazanir** [Aircoookie/WLED#3256]
I want to control led strips thanks to WLED with mbi6033, a chip that was added to the NPB library some time ago.

It is a chip that uses mbi6033 dta+clk, added to the neopixelbus library. I’ve come to a certain point by making some changes to the wled code. But I can’t solve flickering and empty color bytes.

mbi6033 asks for header information first and then uses 48 bit color. In the form of G 16 + B16 + R16. However, the first large byte of this 16-bit is always empty. I added in the picture.

The method we used in the bus_wrapper section was added this way.
`#define B_HS_DOT_3 NeoPixelBusLg<NeoRbg48Feature, Mbi6033Method, NeoGammaNullMethod> `

The data shown in the image below represent: Color = White , brightness = full (from wled control panel)
![16bit4 white data](https://github.com/Aircoookie/WLED/assets/27890571/3744894a-2304-4185-b8a6-8f63f69100c3)

The data shown in the image below represent:  single color, full brightness
![16bit3](https://github.com/Aircoookie/WLED/assets/27890571/17ebb72b-7fef-4a0a-9bff-0e9329071e22)

**Comment by blazoncek** [comment]
*Some time* is 2 days as of this writing. 😀  It will not make it into WLED until new NeoPixelBus is released.
And even then it may take time. You may accelerate the process by sending sample LED strip.

**Comment by seyfettinkazanir** [comment]
Are you talking about sending a physical led strip?

**Comment by blazoncek** [comment]
yes

**Comment by seyfettinkazanir** [comment]
Where do I need to send it, can you send an address?

**Comment by seyfettinkazanir** [comment]
@blazoncek I want to send led strip prepared with mbi6033, how should I do?

**Comment by blazoncek** [comment]
I am currently too busy to tackle this implementation. Perhaps @softhack007 or @Aircoookie are willing to take on the challenge?

If not, you are still free to implement additional LED chip support in your fork and make a PR upstream. You will need to update (at least) the following files:
- settings_leds.htm
- bus_wrapper.h
- const.h

As a base take UCS890x support.

**Comment by seyfettinkazanir** [comment]
I would be very grateful if anyone can help.  I may need all the help. I can send in physical led strip if needed.

settings_leds.htm
bus_wrapper.h
const.h

I tried to make the edits you mentioned. It's a little dim but I couldn't figure out the flickering. I edited it a bit with the mbi6033method prepared in the NPB library. Would it be helpful in terms of help if I stated the edits I made here?

**Comment by softhack007** [comment]
I could take it, but not this month - mid July would be earliest possible for me to start thinking about a solution.

Actually there is a known bug in NeoPixelBus regarding 16bit byte ordering ("endianness") that is only solved in the dev branch of makuna. So it would be smart to wait until a new NPB release is availeable.

* https://github.com/Makuna/NeoPixelBus/issues/700

**Comment by softhack007** [comment]
> couldn't figure out the flickering.

I've seen several people reporting unexpected behaviour of the brightness limiter in latest WLED. Maybe this also causes the flickering you experience?

**Comment by seyfettinkazanir** [comment]
> I could take it, but not this month - mid July would be earliest possible for me to start thinking about a solution.
> 
> Actually there is a known bug in NeoPixelBus regarding 16bit byte ordering ("endianness") that is only solved in the dev branch of makuna. So it would be smart to wait until a new NPB release is availeable.
> 
> * [ColorFeatures using 16bit color elements has endian issue with buffer copies to color objects Makuna/NeoPixelBus#700](https://github.com/Makuna/NeoPixelBus/issues/700)


I guess it looks like a new NPB version is available.  https://github.com/Makuna/NeoPixelBus/releases/tag/2.7.6 

Would you have time to add the Mbi6033 if it won't tire you out?

**Comment by blazoncek** [comment]
> Would you have time to add the Mbi6033 if it won't tire you out?

Last NPB update caused headaches we are still recovering from. :)

**Comment by seyfettinkazanir** [comment]
@blazoncek  Still developing headaches with NPB latest update? I wanted to ask if it's the right time to work on the mbi6033.

**Comment by blazoncek** [comment]
Let's say it is scheduled for 0.15 release. Ok?
😄

**Comment by seyfettinkazanir** [comment]
> 

Thanks, I'm looking forward to it.

**Comment by seyfettinkazanir** [comment]
> Let's say it is scheduled for 0.15 release. Ok? 😄

Greetings, have you had the opportunity to do any work on the Mbi6033? This product excites me a lot. I'm looking forward to your news. Could a schedule be specified for 0.15 release?

**Comment by softhack007** [comment]
> > Let's say it is scheduled for 0.15 release. Ok? 😄
> 
> Could a schedule be specified for 0.15 release?

Considering that all developers are doing it in their spare time (hobby), it's very hard to give any dates. 😉 "it's ready once we consider it well-done".

I think all we can say today is
* 0.14.0 is still bet and not ready to be released
* 0.15.0 will come after 0.14.x

**Comment by seyfettinkazanir** [comment]
Hello, has there been any progress regarding mbi6033 for 15.0? We look forward to any developments in this regard.

@softhack007 @blazoncek

**Comment by blazoncek** [comment]
No. But you can submit PR. 0.15 as of now uses NePixelBus 2.7.8 which is the latest release available ATM.
